### DISCUSSION TITLE
How to clear a Date Filter shown date?

### DISCUSSION THREAD

**Discussion by jorgemoralespou** [KevinVandy/material-react-table#554] [Q&A] [answered]
Looking at the example on https://www.material-react-table.com/docs/examples/advanced when the date filter on last column is clear after it's been selected and a date input, the date is not cleared. How can the clear filter on the column action be propagated to the DatePicker component so that the date is cleared?

**Comment by joey-kwl** [comment, marked as answer]
I added the clear button myself

```
const ClearButton = ({ column }: { column: MRT_Column }) => {
	return (
		<Tooltip title="Clear filter" arrow>
			<IconButton
				size="small"
				aria-label="Clear filter"
				onClick={(e) => { e.preventDefault(); column.setFilterValue(undefined) }}
				style={{ order: 1 }}>
				<ClearIcon color="disabled" fontSize="small" />
			</IconButton>

		</Tooltip>
	)

}

<DatePicker
	onChange={(newValue) => {
		column.setFilterValue(newValue);
	}}
	slotProps={{
		actionBar: {
			actions: ["clear"],
		},
		textField: {
			variant: "standard",
			helperText: `Filter Mode: ${filterName}`,
			sx: {
				width: "100%"
			},
			InputProps: {
				startAdornment: <ClearButton column={column} />
			},

		},
		openPickerButton: {
			sx: {
				marginRight: 0
			},
			size: "small"
		}
	}}
	format={"DD-MM-YYYY"}
	value={column.getFilterValue() ?? null}
/>
```

![image](https://github.com/KevinVandy/material-react-table/assets/10912994/cfcebe99-45d0-449a-80dc-a4e5f4d195c0)

**Reply by jorgemoralespou** [reply]
This is cool. Does it clear when you click clear filters in the kebab menu?

**Reply by joey-kwl** [reply]
If you are talking about this one, then yes, it also clear the datepicker

![image](https://github.com/KevinVandy/material-react-table/assets/10912994/0ae3fae3-ce53-47ec-a5e5-9c5bcc089d41)

**Reply by jorgemoralespou** [reply]
Thanks so much. Work great!
